### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by rmarren1**
Just noticed we did not have one of these for Dash. I added a few checklists for our own use, I use these before making any permanent changes like merging a PR and releasing a package

I added the `git tag` step since a few people in the community have requested we do this.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+- [ ] You have updated the `dash/version.py` file and the top of `CHANGELOG.md`
+- [ ] Two people have :dancer:'d the pull request. You can be one of these people if you are a Dash core contributor.
+
+### Pre-Release checklist
+- [ ] Everything in the Pre-Merge checklist is completed. (Except the last one if this is a release candidate).
+- [ ] `git remote show origin` shows you are in the correct repository.
+- [ ] `git branch` shows that you are on the expected branch.
+- [ ] `git status` shows that there are no unexpected changes.
+- [ ] `dash/version.py` is at the correct version.
+- [ ] You have tagged the release using `git tag v<version_number`.
```
Comment (T4rk1n) [inline CONTRIBUTING.md]: I think tagging should be done after the merge on the merge commit. Maybe we could setup circle to add a tag on merge and `version.py` has changed.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+Or:
+`nose2 -v`
+
+## Local configuration
+You can configure the test server with the following variables:
+### DASH_TEST_CHROMEPATH
+If you run a special chrome set the path to your chrome binary with this environment variable.
+
+### DASH_TEST_PROCESSES
+If you encounter errors about Multi-server + Multi-processing when running under Python 3 try running the tests with the number of server processes set to 1.
```
Comment (T4rk1n) [inline CONTRIBUTING.md]: The tests also won't run on windows. Maybe we could change the `IntegrationTests` in this repo with the one in dash-core-components, it runs threaded instead when windows is detected.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    * In the description, link to the PR and any relevant issue(s)
+    * Pin the topic so that it appears at the top of the forum for two weeks
+
+## [Checklists](http://rs.io/unreasonable-effectiveness-of-checklists/)
+### Pre-Merge checklist
+- [ ] All tests on CircleCI have passed.
+- [ ] All visual regression differences have been approved.
+- [ ] If changes are significant, a release candidate has been created and posted to Slack, the Plotly forums, and at the very top of the pull request.
+- [ ] You have updated the `dash/version.py` file and the top of `CHANGELOG.md`
+- [ ] Two people have :dancer:'d the pull request. You can be one of these people if you are a Dash core contributor.
```
Comment (T4rk1n) [inline CONTRIBUTING.md]: Maybe add the list of reviewers we have on slack for this repo which is me and you.

**Comment by rmarren1** [comment]
I fixed those, just made a post-merge and post-release checklist that can be expanded upon later (they are one item right now).

I think a CI hook would be good for git tagging. I'm not sure how / if we can trigger something to happen when a PR is merged though. I'll do some more research.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+Please lint any additions to react components with `pylint` and `flake8`.
+
+## Pull Request Guidelines
+
+Use the [GitHub flow][] when proposing contributions to this repository (i.e. create a feature branch and submit a PR against the master branch).
+
+## Running the Tests
+
+**Warning:** _Tests do not currently run on windows._
```
Comment (chriddyp) [inline CONTRIBUTING.md]: could we create an issue about this and then link the issue inside this `CONTRIBUTING.md` file? that way, a windows contributor has a place to ask us about how to get past this

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
+# Contributing to dash
+
+## Getting Started
+
+Refer to the [readme](README.md) for installation and development instructions.
+
+## Coding Style
+
+Please lint any additions to react components with `pylint` and `flake8`.
```
Comment (chriddyp) [inline CONTRIBUTING.md]: "to react components"? do you just mean to python code?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+If you encounter errors about Multi-server + Multi-processing when running under Python 3 try running the tests with the number of server processes set to 1.
+
+### Example: single test run with configuration
+```
+DASH_TEST_CHROMEPATH=/bin/google-chrome-beta DASH_TEST_PROCESSES=1 python -m unittest -v test.test_integration.Tests.test_inputs
+```
+
+## Making a contribution
+
+1. Create a pull request and tag the Plotly team (`@plotly/dash`) and tag / request review from [@rmarren1](https://github.com/rmarren1) and [@T4rk1n](https://github.com/T4rk1n).
```
Comment (chriddyp) [inline CONTRIBUTING.md]: Could we add a step like: "For larger features, you're contribution will have a higher likelihood of getting merged into this repository if you create an issue to discuss the changed that you'd like to make before you create a pull request."

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+- [ ] `git branch` shows that you are on the expected branch.
+- [ ] `git status` shows that there are no unexpected changes.
+- [ ] `dash/version.py` is at the correct version.
+
+### Post-Release checklist
+- [ ] You have closed all issues that this pull request solves, and commented the new version number users should install.
+
+## Financial Contributions
+
+If your company wishes to sponsor development of open source dash components, please [get in touch][].
```
Comment (chriddyp) [inline CONTRIBUTING.md]: Let's say "Dash, and many of Plotly's open source products, have been funded through direct sponsorship by companies. [Get in touch] about funding feature additions, consulting, or custom app development."

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    * Title it `":mega: Announcement! New <Your Feature> - Feedback Welcome"`
+    * In the description, link to the PR and any relevant issue(s)
+    * Pin the topic so that it appears at the top of the forum for two weeks
+
+## [Checklists](http://rs.io/unreasonable-effectiveness-of-checklists/)
+### Pre-Merge checklist
+- [ ] All tests on CircleCI have passed.
+- [ ] All visual regression differences have been approved.
+- [ ] If changes are significant, a release candidate has been created and posted to Slack, the Plotly forums, and at the very top of the pull request.
+- [ ] You have updated the `dash/version.py` file and the top of `CHANGELOG.md`
```
Comment (chriddyp) [inline CONTRIBUTING.md]: Can we have something like "For larger additions, your CHANGELOG entry includes sample code about how the feature works. The feature should also link to the original pull request(s)."

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+- [ ] Two people have :dancer:'d the pull request. You can be one of these people if you are a Dash core contributor.
+
```
Comment (chriddyp) [inline CONTRIBUTING.md]: Can we add a new item like "- [ ] You have created an issue about documenting the new feature or change and you have added it to the Documentation project."

Comment (chriddyp) [reply CONTRIBUTING.md]: sorry, that would go into the "Post-release checklist"

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+### Pre-Release checklist
+- [ ] Everything in the Pre-Merge checklist is completed. (Except the last two if this is a release candidate).
+- [ ] `git remote show origin` shows you are in the correct repository.
+- [ ] `git branch` shows that you are on the expected branch.
+- [ ] `git status` shows that there are no unexpected changes.
+- [ ] `dash/version.py` is at the correct version.
+
+### Post-Release checklist
+- [ ] You have closed all issues that this pull request solves, and commented the new version number users should install.
```
Comment (chriddyp) [inline CONTRIBUTING.md]: in addition to a documentation issue, can we also add "- [ ] You have created a pull request in `dash-docs` with the new release of your feature by editing that project's [`requirements.txt` file](https://github.com/plotly/dash-docs/blob/master/requirements.txt) and you have assigned `@chriddyp` to review.

**Comment by rmarren1** [comment]
@chriddyp Made those changes!

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+## Making a contribution
+_For larger features, your contribution will have a higher likelihood of getting merged if you create an issue to discuss the changes that you'd like to make before you create a pull request._
+
+1. Create a pull request and tag the Plotly team (`@plotly/dash`) and tag / request review from [@rmarren1](https://github.com/rmarren1) and [@T4rk1n](https://github.com/T4rk1n).
+2. After a review has been done and your changes have been approved, create a prerelease and comment in the PR. Version numbers should follow [semantic versioning][]. To create a prerelease:
+    * Add `rc1` to `version.py` (`./dash/version.py`) e.g. `0.13.0rc1`
+        - If needed, ask @chriddyp to get PyPi package publishing access.
+    * Run `python setup.py sdist` to build a distribution zip.
+    * Check the `dist` folder for a zip ending with your selected version number. Double check that this version number ends with `rc#`, as to not mistakenly publish the package.
```
Comment (T4rk1n) [inline CONTRIBUTING.md]: It's a `tar` file, not a `zip`.

**Comment by T4rk1n** [comment]
:dancer:

**Pull request merged**